client/allocrunner/taskrunner/envoybootstrap_hook_test.go

```diff
@@ -17,13 +17,222 @@ import (
 	agentconsul "github.com/hashicorp/nomad/command/agent/consul"
 	"github.com/hashicorp/nomad/helper/args"
 	"github.com/hashicorp/nomad/helper/testlog"
+	"github.com/hashicorp/nomad/helper/uuid"
 	"github.com/hashicorp/nomad/nomad/mock"
 	"github.com/hashicorp/nomad/nomad/structs"
 	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/unix"
 )
 
 var _ interfaces.TaskPrestartHook = (*envoyBootstrapHook)(nil)
 
+func writeTmp(t *testing.T, s string, fm os.FileMode) string {
+	dir, err := ioutil.TempDir("", "envoy-")
+	require.NoError(t, err)
+
+	fPath := filepath.Join(dir, sidsTokenFile)
+	err = ioutil.WriteFile(fPath, []byte(s), fm)
+	require.NoError(t, err)
+
+	return dir
+}
+
+func TestEnvoyBootstrapHook_maybeLoadSIToken(t *testing.T) {
+	t.Parallel()
+
+	if unix.Geteuid() == 0 {
```

Comment:
I think this will break the build on Windows. Since Windows doesn't support Connect yet, feel free to add a build tag to exclude this file on Windows.

On the other hand: why doesn't this test work as root?!

Reply:
The `file is unreadable` subtest fails, because it emulates an unreadable file by setting permissions to `0200`. As a normal user that works great, but the root user is able to read the file regardless of the permissions.

Ideally I think it'd be nice to be using an interface of the filesystem (e.g. https://github.com/src-d/go-billy) which we could then mock out and simulate all kinds of FS error conditions, but that would involve a pretty extensive refactoring across `nomad/`